app/models/note.rb

```diff
@@ -31,7 +31,10 @@ class Note < ApplicationRecord
   belongs_to :page, optional: true
   belongs_to :work, optional: true
   belongs_to :collection, optional: true
-  has_one :deed, :dependent => :destroy
+
+  has_many :deeds, dependent: :destroy
+  has_one :deed, -> { order(created_at: :desc) }
```

Comment:
Played with this a bit more. It seems that doing `has_one :deed` is only fetching the first deed linked to the current note (deed.note_id == note.id)

If for some reason we created multiple deeds for current note, it will only ever get the first deed created.

To avoid orphaning the other deeds linked, we added `has_many :deeds, dependent: :destroy` so that all deeds linked to a note is deleted. We still kept the `has_one :deed` relationship so functionality is retained (except that I scoped it to get the LATEST deed. Let me know if we should always get the FIRST deed linked)

Tested and it successfully notes without orphaning deeds.
@benwbrum

Reply:
Isn't this a natural feature of `has_one` associations?  Unless there's a `UNIQUE` constraint on the foreign key column in the database, there is no way to actually enforce a single child record per parent.

Reply:
It's just that we have lines like these
  def record_deed
    deed = Deed.new
    deed.note = @note # here
    deed.page = @page
    deed.work = @work
    deed.collection = @work.collection
    deed.deed_type = DeedType::NOTE_ADDED
    deed.user = current_user

    deed.save!
    update_search_attempt_contributions
  end
Can we assure that we are not creating multiple deed for a single note? That would then fall to the described behaviour I mentioned in comment @benwbrum 

As per enforcing single child record, we are still doing what we initially have before, just that I changed it so it checks in the last element instead of the first (which is not going to affect anyway if we indeed cannot make 2 deeds have the same parent note_id).

So really the only change is that, we allow all related deeds to be deleted on note destroy.